chaossearch/resource_object_group.go

```diff
@@ -98,9 +106,17 @@ func resourceObjectGroup() *schema.Resource {
 	}
 }
 
+
+
 func resourceObjectGroupCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	c := meta.(*ProviderMeta).Client
 
+	array_flatten_tf := data.Get("array_flatten_depth").(int)
```

Comment:
To keep this function purely terraform-ish, I'd suggest to move this logic into a separate function as you did it initially

Reply:
I'm not sure this makes it clearer, we have a parallel piece of logic in the "read" method